typing/typemod.ml

```diff
@@ -2769,84 +2768,85 @@ let gen_annot outputprefix sourcefile annots =
+        normalize_signature simple_sg;
+        Typecore.force_delayed_checks ();
+        (* See comment above. Here the target signature contains all
+           the value being exported. We can still capture unused
+           declarations like "let x = true;; let x = 1;;", because in this
+           case, the inferred signature contains only the last declaration. *)
+        if not !Clflags.dont_write_files then begin
+          let alerts = Builtin_attributes.alerts_of_str ast in
+          let cmi =
+            Env.save_signature ~alerts
+              simple_sg modulename (outputprefix ^ ".cmi")
           in
-          Typecore.force_delayed_checks ();
-          (* It is important to run these checks after the inclusion test above,
-             so that value declarations which are not used internally but
-             exported are not reported as being unused. *)
           let annots = Cmt_format.Implementation str in
-          Cmt_format.save_cmt (outputprefix ^ ".cmt") modulename
-            annots (Some sourcefile) initial_env None;
-          gen_annot outputprefix sourcefile annots;
-          { structure = str;
-            coercion;
-            signature = dclsig
-          }
-        end else begin
-          Location.prerr_warning (Location.in_file sourcefile)
-            Warnings.Missing_mli;
-          let coercion =
-            Includemod.compunit initial_env ~mark:Mark_positive
-              sourcefile sg "(inferred signature)" simple_sg
-          in
-          check_nongen_schemes finalenv simple_sg;
-          normalize_signature simple_sg;
-          Typecore.force_delayed_checks ();
-          (* See comment above. Here the target signature contains all
-             the value being exported. We can still capture unused
-             declarations like "let x = true;; let x = 1;;", because in this
-             case, the inferred signature contains only the last declaration. *)
-          if not !Clflags.dont_write_files then begin
-            let alerts = Builtin_attributes.alerts_of_str ast in
-            let cmi =
-              Env.save_signature ~alerts
-                simple_sg modulename (outputprefix ^ ".cmi")
-            in
-            let annots = Cmt_format.Implementation str in
-            Cmt_format.save_cmt  (outputprefix ^ ".cmt") modulename
-              annots (Some sourcefile) initial_env (Some cmi);
-            gen_annot outputprefix sourcefile annots
-          end;
-          { structure = str;
-            coercion;
-            signature = simple_sg
-          }
-        end
+          Cmt_format.save_cmt  (outputprefix ^ ".cmt") modulename
+            annots (Some sourcefile) initial_env (Some cmi);
+          gen_annot outputprefix sourcefile annots
+        end;
+        { structure = str;
+          coercion;
+          signature = simple_sg
+        }
       end
-    )
+    end
+  in
+  Misc.try_finally (fun () -> Builtin_attributes.warning_scope [] run)
```

Comment:
Review note: all changes in `type_implementation` above that line are the result of an change of indentations.